ako-gateway-api/nodes/gateway_model_rel.go

```diff
@@ -224,8 +224,18 @@ func GatewayToRoutes(namespace, name, key string) ([]string, bool) {
 			}
 		}
 	}
+	allRouteTypeNsNameList := []string{}
 	routeTypeNsNameList, _ := validateReferredHTTPRoute(key, name, namespace, allowedRoutes)
-	return routeTypeNsNameList, true
+	for _, routeTypeNsName := range routeTypeNsNameList {
+		_, namespace, name := lib.ExtractTypeNameNamespace(routeTypeNsName)
+		routes, found := HTTPRouteChanges(namespace, name, key)
```

Comment:
@abhishekbsingh: Instead of doing this, we change `HTTPRouteChanges` function to map service to Gateway and service to HTTPRoute mapping, it will avoid double processing.
Currently we just store Gateway to service mapping and not reverse.
https://github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/blob/master/ako-gateway-api/nodes/gateway_model_rel.go#L490-L494 --> Here we can add reverse mapping 

same for HTTPRoute to service mapping present at https://github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/blob/master/ako-gateway-api/nodes/gateway_model_rel.go#L485-L487

we may need to add some store function for this.